src/main/java/kitchenpos/tablegroup/application/TableGroupService.java

```diff
@@ -0,0 +1,54 @@
+package kitchenpos.tablegroup.application;
+
+import kitchenpos.order.domain.OrderRepository;
+import kitchenpos.order.domain.OrderStatus;
+import kitchenpos.table.OrderTableRepository;
+import kitchenpos.table.OrderTable;
+import kitchenpos.tablegroup.domain.TableGroup;
+import kitchenpos.tablegroup.domain.TableGroupRepository;
+import kitchenpos.tablegroup.dto.TableGroupAddRequest;
+import kitchenpos.tablegroup.dto.TableGroupMapper;
+import kitchenpos.tablegroup.dto.TableGroupResponse;
+import org.mapstruct.factory.Mappers;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@Service
+public class TableGroupService {
+    private final OrderRepository orderDao;
+    private final OrderTableRepository orderTableRepository;
+    private final TableGroupRepository tableGroupRepository;
+    private final TableGroupMapper mapper = Mappers.getMapper(TableGroupMapper.class);
+
+    public TableGroupService(final OrderRepository orderDao, final OrderTableRepository orderTableRepository, final TableGroupRepository tableGroupRepository) {
+        this.orderDao = orderDao;
+        this.orderTableRepository = orderTableRepository;
+        this.tableGroupRepository = tableGroupRepository;
+    }
+
+    @Transactional
+    public TableGroupResponse create(final TableGroupAddRequest request) {
+        final List<OrderTable> orderTables = request.getOrderTables().stream()
+                .map(it -> orderTableRepository.getOne(it.getId()))
+                .collect(Collectors.toList());
+        request.checkSameSize(orderTables.size());
+        return mapper.toResponse(tableGroupRepository.save(new TableGroup(orderTables)));
+    }
+
+    @Transactional
+    public void ungroup(final Long tableGroupId) {
+        TableGroup tableGroup = tableGroupRepository.getOne(tableGroupId);
+        final List<Long> orderTableIds = tableGroup.getOrderTables().stream()
+                .map(OrderTable::getId)
+                .collect(Collectors.toList());
+        if (orderDao.existsByOrderTable_IdInAndOrderStatusIn(orderTableIds, Arrays.asList(OrderStatus.COOKING, OrderStatus.MEAL))) {
+            throw new IllegalArgumentException();
+        }
+        tableGroup.getOrderTables().forEach(OrderTable::ungroup);
```

Comment:
도메인 값을 꺼내서 실행하지 않고 도메인에게 실행을 위임하면 더 객체지향스러운 코드가 되겠네요

```suggestion
        tableGroup.ungroup();
```